src/mpid/ch4/netmod/ucx/ucx_pre.h

```diff
@@ -58,7 +58,7 @@ typedef struct MPIDI_UCX_win_target_sync {
 } MPIDI_UCX_win_target_sync_t;
 
 typedef struct {
-    MPIDI_UCX_win_info_t *info_table;
+    MPIDI_UCX_win_info_t *info_table;   /* NULL indicates AM fallback for entire win */
```

Comment:
I don't quite like adding extra purpose to a variable. It makes the code more obscure to read. Why not add and use a separate variable, e.g. `bool none_reachable`?

Reply:
It seems straightforward to me that if per-target `info_table` is not set it means all AM fallback. I don't really think a separate flag is needed.